tfjs-backend-webgpu/src/kernels/conv2d_mm_webgpu.ts

```diff
@@ -109,41 +113,35 @@ export class Conv2DMMProgram implements WebGPUProgram {
         ${matMulSource}
 
         int batch;
-
+        int dimAOuter = outShape[1] * outShape[2];
+        int dimBOuter = outShape[3];
+        int dimInner = filterDims[0] * filterDims[1] * xShape[3];
         float mm_readA(int row, int col) {
           int r = int(row), c = int(col);
-          ivec4 coord = ivec4(
-              (c / filterDims[1]) % filterDims[0],
-              c % filterDims[1],
-              c / (filterDims[0] * filterDims[1]),
-              r);
-
-          ivec4 shape = ivec4(filterDims, xShape[3], outShape[3]);
-          return ${sampleA};
-        }
-
-        float mm_readB(int row, int col) {
-          int r = int(row), c = int(col);
-          int outRow = c / outShape[2];
-          int outCol = c % outShape[2];
+          int outRow = r / outShape[2];
+          int outCol = r % outShape[2];
 
-          int WRow = (r / filterDims[1]) % filterDims[0];
-          int WCol = r % filterDims[1];
+          int WRow = c / (filterDims[1] * xShape[3]);
```

Comment:
Can you splice in `${filterDims[1] * xShape[3]}` as a constant here as well?

Reply:
I understand that making it as constant can save once calculation. But it will results we have variant shaders which will increase the compilation time of shader in the warm up stage. Maybe this overhead can be ignored?